test/PreAMMBatcher.spec.ts

```diff
@@ -171,16 +165,21 @@ describe("PreAMMBatcher", () => {
       testCaseInput.sellOrdersToken1[0],
     );
 
-    await batcher.batchTrade(
-      testCaseInput.sellOrdersToken0[0].encode(),
-      testCaseInput.sellOrdersToken1[0].encode(),
-      { gasLimit: 6000000 },
-    );
-    expect("transferFrom").to.be.calledOnContractWith(token0, [
-      walletTrader1.address,
-      batcher.address,
-      testCaseInput.sellOrdersToken0[0].sellAmount.toString(),
-    ]);
+    await token0.mock.transferFrom.reverts();
+    await token0.mock.transferFrom
+      .withArgs(
+        walletTrader1.address,
+        batcher.address,
+        testCaseInput.sellOrdersToken0[0].sellAmount,
+      )
+      .returns(true);
+    await expect(
```

Comment:
I am not sure whether I like this way of testing.

if by accident I remove the transferFrom completely out of the smart contract, and transferFrom is never called, then also the "expect" statement will pass, as there is no revert. Only e2e tests would catch the issue.

Before, we have tested: transferFrom is called with parameters x, now we are testing: If transferFrom was called, then it had the right parameters.

I am fine with keeping this PRs as it is, but longterm investigating for a better solution would be great.

Reply:
Hmm, good point. I'll try to come up with something.

Reply:
I think removing the transfer from the smart contract should be caught by a test that verifies `batchTrade` fails if transfer fails.

Reply:
This is already being tested :relieved: